src/components/SigninModal/MainContent/SigninForm/SigninForm.jsx

```diff
@@ -0,0 +1,165 @@
+        userName: '',
+        email: '',
+        password: '',
+        rePassword: ''
+    });
+
+    const handleOnChange = (e) => {
+        setUser({
+            ...user,
+            [e.target.id]: e.target.value
+        })
+    }
+
+    const handleSignIn = async (e) => {
+        e.preventDefault();
+        try {
+            await firebase.firestore().collection('users').add({
+                    userName: user.userName,
+                    email: user.email
+                })
+            await firebase.auth().createUserWithEmailAndPassword(user.email, user.password);
+        } catch (err){
+            console.log(err)
+        }
+    }
+    return (
+        <form onSubmit={handleSignIn}>
+            <FormContainer>
+                <InputField>
+                    <InputLabelContainer>
+                        <label htmlFor="userName">
+                            <InputLabelText>Username</InputLabelText>
+                        </label>
+                    </InputLabelContainer>
+                    <InputContainer>
+                        <Input type="userName" 
+                               id="userName" 
+                               size="52" 
+                               onChange={handleOnChange} 
+                               value={user.userName} 
+                                />
+                    </InputContainer>
+                </InputField>
+                <InputField>
+                    <InputLabelContainer>
+                        <label htmlFor="email">
+                            <InputLabelText>Email</InputLabelText>
+                        </label>
+                    </InputLabelContainer>
+                    <InputContainer>
+                        <Input type="email" 
+                               id="email" 
+                               size="52" 
+                               onChange={handleOnChange} 
+                               value={user.email} 
+                                />
+                    </InputContainer>
+                </InputField>
+                <InputField>
+                    <InputLabelContainer htmlFor="password">
+                        <InputLabelText>Password</InputLabelText>
+                    </InputLabelContainer>
+                    <InputContainer>
+                        <Input type="password" 
+                               id="password"
+                               size="52" 
+                               onChange={handleOnChange} 
+                               value={user.password} 
+                               />
+                    </InputContainer>
+                </InputField>
+                <InputField>
+                    <InputLabelContainer htmlFor="rePassword">
+                        <InputLabelText>Re-ender Password</InputLabelText>
+                    </InputLabelContainer>
+                    <InputContainer>
+                        <Input type="rePassword" 
```

Comment:
there is no such type `rePassword`. should be `password` 👍 
https://www.w3schools.com/html/html_form_input_types.asp